code/src/layouts/header/components/header-toolbar/HeaderToolbar.tsx

```diff
@@ -77,21 +63,64 @@ const HeaderToolbar = () => {
     openDrawer(<CartDrawer />);
   };
 
-  const cartItemsCount = data?.items.reduce(
-    (total, item) => total + item.quantity,
-    0
+  const isUser = ROLES.USER === userRole;
+  const isShopManager = ROLES.SHOP_MANAGER === userRole;
+  const isAdmin = ROLES.ADMIN === userRole;
+
+  const dashboardButton = (isShopManager || isAdmin) && (
+    <AppTooltip titleTranslationKey="dashboard.tooltip" key="dashboard-button">
+      <AppIconButton
+        to={routes.dashboard.orders.path}
+        component={AppLink}
+        data-testid="dashboard-button"
+        data-cy="dashboard-button"
+      >
+        <DashboardCustomizeIcon fontSize="medium" />
+      </AppIconButton>
+    </AppTooltip>
   );
 
-  const displayedCartItemsCount =
-    cartItemsCount && cartItemsCount > 99 ? "99+" : cartItemsCount;
+  const ordersButton = isUser && (
+    <AppTooltip titleTranslationKey="orders.tooltip" key="orders-button">
+      <AppIconButton
+        to={routes.orders.path}
+        component={AppLink}
+        data-cy="orders-button"
+      >
+        <ListAltIcon className="header__toolbar-icon" fontSize="medium" />
+      </AppIconButton>
+    </AppTooltip>
+  );
+
+  const authButton = isAuthenticated ? (
+    <AppButton
+      onClick={handleLogout}
+      variant="danger"
+      size="small"
+      key="auth-button"
+      data-testid="LogoutButton"
+    >
+      <LogoutIcon />
+    </AppButton>
+  ) : (
+    <AppButton
+      onClick={handleOpenAuthModal}
+      key="auth-button"
+      data-cy="auth-button"
+    >
+      <AppTypography component="span" translationKey="signIn.label" />
+    </AppButton>
+  );
+
+  const cartItemsCount = calculateCartItemsCounter(data.items);
```

Comment:
I think header component became too complicated to understand. Maybe we can create three separate toolbars for different roles? I think it would be easier to maintain, especailly if we will have more roles in the future. Also then we won't need to import unnesessary components or files if they are not needed at all. We can use lazy imports here then

Reply:
Thanks for suggestion. I divided that component into components according to existing roles. Now it looks much better, because that component was a mess. You also suggested use of lazy imports, but I am not sure about it. It worsens user experience because loaders appear when user lands on a website and when he logs in. It looks not nice.